homeassistant/components/ridwell/config_flow.py

```diff
@@ -0,0 +1,132 @@
+"""Config flow for Ridwell integration."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from aioridwell import async_get_client
+from aioridwell.errors import InvalidCredentialsError, RidwellError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, LOGGER
+
+STEP_REAUTH_CONFIRM_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WattTime."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        self._password: str | None = None
+        self._reauthing: bool = False
+        self._username: str | None = None
+
+    @callback
+    def _async_show_errors(
+        self, errors: dict, error_step_id: str, error_schema: vol.Schema
+    ) -> FlowResult:
+        """Show an error on the correct form."""
+        return self.async_show_form(
+            step_id=error_step_id,
+            data_schema=error_schema,
+            errors=errors,
+            description_placeholders={CONF_USERNAME: self._username},
+        )
+
+    async def _async_validate(
+        self, error_step_id: str, error_schema: vol.Schema
+    ) -> FlowResult:
+        """Validate input credentials and proceed accordingly."""
+        session = aiohttp_client.async_get_clientsession(self.hass)
+
+        if TYPE_CHECKING:
+            assert self._password
+            assert self._username
+
+        try:
+            await async_get_client(self._username, self._password, session=session)
+        except InvalidCredentialsError:
+            return self._async_show_errors(
+                {"base": "invalid_auth"}, error_step_id, error_schema
+            )
+        except RidwellError as err:
+            LOGGER.error("Unknown Ridwell error: %s", err)
+            return self._async_show_errors(
+                {"base": "unknown"}, error_step_id, error_schema
+            )
+
+        if self._reauthing:
```

Comment:
Do we need this check? Is it possible to be in a reauth flow without having an existing username set as unique id?